### PR TITLE
docs(core): add public API for standalone entities

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
This commit adds public API docs for standalone components,
directives and pipes.

### CODE REVIEW

**Review by alxhub — DISMISSED** [review]

File: packages/core/src/metadata/directives.ts [hunk 1]
```diff
@@ -557,11 +561,31 @@ export interface Component extends Directive {
...
   standalone?: boolean;
 
-  // TODO: better type here!
+  /**
+   * The imports property specifies the standalone component's template dependencies — those
+   * directives, components, and pipes that can be used within its template. Standalone components
+   * can import other standalone components, directives and pipes as well as existing NgModules.
+   *
+   * This property is only available for standalone components - specifying it for components
+   * declared in NgModule is a compilation error.
```
Comment (dario-piotrowicz) [inline packages/core/src/metadata/directives.ts:578]:
```suggestion
   * declared in an NgModule generates a compilation error.
```

File: packages/core/src/metadata/directives.ts [hunk 2]
```diff
@@ -557,11 +561,31 @@ export interface Component extends Directive {
...
+   */
   imports?: (Type<any>|any[])[];
 
+  /**
+   * The set of schemas that declare elements to be allowed in a standalone component. Elements and
+   * properties that are neither Angular components nor directives must be declared in a schema.
+   *
+   * This property is only available for standalone components - specifying it for components
+   * declared in NgModule is a compilation error.
```
Comment (dario-piotrowicz) [inline packages/core/src/metadata/directives.ts:587]:
```suggestion
   * declared in an NgModule generates a compilation error.
```

File: packages/core/src/metadata/directives.ts [hunk 3]
```diff
@@ -557,11 +561,31 @@ export interface Component extends Directive {
    */
   preserveWhitespaces?: boolean;
 
+  /**
+   * Angular components marked as `standalone` do not need to be declared in an NgModule. Such
+   * components directly manage their own template dependencies (components, directives and pipes
+   * used in a template) and don't depend on any "intermediate context" of a NgModule (ex.
```
Comment (dario-piotrowicz) [inline packages/core/src/metadata/directives.ts:567]:
```suggestion
   * used in a template) and don't depend on any "intermediate context" of an NgModule (ex.
```

File: packages/core/src/metadata/directives.ts [hunk 4]
```diff
@@ -635,6 +659,10 @@ export interface Pipe {
    */
   pure?: boolean;
 
+  /**
+   * Angular pipes marked as `standalone` do not need to be declared in an NgModule. Such
+   * pipes don't depend on any "intermediate context" of a NgModule (ex. configured providers).
```
Comment (dario-piotrowicz) [inline packages/core/src/metadata/directives.ts:664]:
```suggestion
   * pipes don't depend on any "intermediate context" of an NgModule (ex. configured providers).
```

**Review by AndrewKushnir — DISMISSED** [review]

**Review by jessicajaniuk — COMMENTED** [review]
reviewed-for: public-api

File: packages/core/src/metadata/directives.ts [hunk 5]
```diff
@@ -289,6 +289,10 @@ export interface Directive {
    */
   jit?: true;
 
+  /**
+   * Angular directives marked as `standalone` do not need to be declared in an NgModule. Such
+   * directives don't depend on any "intermediate context" of a NgModule (ex. configured providers).
```
Comment (dario-piotrowicz) [inline packages/core/src/metadata/directives.ts:294]:
```suggestion
   * directives don't depend on any "intermediate context" of an NgModule (ex. configured providers).
```
:grin:

**Review by dylhunn — APPROVED** [review]
reviewed-for: public-api

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: public-api

**Comment by AndrewKushnir** [comment]
Merge-assistance: it looks like this PR has all the necessary reviews, but Pullapprove check still requires one more `public-api` review (a review from Jess is not taken into account). The PR is ready for merge.

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit 3d5363cc5ea9448fb5157df7dc66e0f77a47c500.

**Pull request closed without merging**